cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPATAuthenticationStrategy.java

```diff
@@ -30,15 +34,53 @@ public class GitPATAuthenticationStrategy implements AuthenticationStrategy {
   private static final String GITHUB_PAT_USERNAME = "oauth2";
 
   @Override
-  public CredentialsProvider getCredentialsProvider(SecureStore store,
-                                                    RepositoryConfig config,
-                                                    String namespaceId) throws AuthenticationConfigException {
-    try {
-      byte[] bytes = store.get(namespaceId, config.getAuth().getTokenName()).get();
-      String token = new String(bytes, StandardCharsets.UTF_8);
-      return new UsernamePasswordCredentialsProvider(GITHUB_PAT_USERNAME, token);
-    } catch (Exception e) {
-      throw new AuthenticationConfigException("Failed to get auth token from secure store", e);
+  public CredentialsProvider getCredentialsProvider(SecureStore store, RepositoryConfig config, String namespaceId) {
+    return new SecureStoreUserNamePasswordCredentialsProvider(store,
+                                                              GITHUB_PAT_USERNAME,
+                                                              config.getAuth().getTokenName(),
+                                                              namespaceId);
+  }
+
+  /**
+   * A class that wraps a {@link UsernamePasswordCredentialsProvider} but fetches password from a {@link SecureStore}.
+   */
+  private static class SecureStoreUserNamePasswordCredentialsProvider extends CredentialsProvider {
```

Comment:
this is really long, can we shorten it to SecureStorePasswordProvider

Reply:
Done.